index.js

```diff
@@ -3,14 +3,18 @@ var recast = require('recast');
 
 var transformer = require('./unreachableBranchTransformer');
 
-module.exports = function (file, opts) {
-  opts = opts || [];
-  var ignore = ['.json'].concat(opts.ignore || []).some(function(ext) {
-    return file.indexOf(ext, file.length - ext.length) !== -1;
-  });
+module.exports = function () {
+  var file = typeof arguments[0] == 'string' ? arguments[0] : null;
+  var opts = (typeof arguments[0] == 'string' ? arguments[1] : arguments[0]) || {};
+
+  if (file){
```

Comment:
When "overloading" functions in js I like to expressively declare what the arguments really are at the very top of the function, separate from the real logic, similarly to how we conventionally declare dependencies at the top of our modules.
Using `function (file, opts) {if (typeof file === 'string') {` we would *normally* (if there were any options to apply when file !== 'string') still need +`opts = opts || {}; /* handle ignore option */ } else { opts = file || {}; }`, which (a) mixes chunks of argument-dependent and argument-declaring code, impacting readability and (b) isn't concise with default opts
That said, since right now the `opts` are only used when file !== 'string', I could optimize this for performance rather than human-readability.
Which of the two concerns you more?